services/frontend-service/src/ui/Pages/Locks/LocksPage.test.tsx

```diff
@@ -40,120 +46,551 @@ describe('LocksPage', () => {
+                            queuedVersion: 0,
+                            undeployVersion: false,
+                            locks: {
+                                locktest: {
+                                    message: 'locktest',
+                                    lockId: 'ui-v2-1337',
+                                    createdAt: new Date(1995, 11, 17),
+                                },
+                                lockbar: { message: 'lockbar', lockId: 'ui-v2-321', createdAt: new Date(1995, 11, 15) },
+                            },
+                        },
+                    },
+                },
+            },
+            filter: 'foo',
+            expectedLockIDs: ['ui-v2-123'],
+        },
+        {
+            name: 'get filtered locks (development, get 2 lock)',
+            envs: {
+                integration: {
+                    name: 'integration',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {
+                        foo: {
+                            name: 'foo',
+                            version: 420,
+                            queuedVersion: 0,
+                            undeployVersion: false,
+                            locks: {
+                                lockfoo: { message: 'lockfoo', lockId: 'ui-v2-123', createdAt: new Date(1995, 11, 16) },
+                            },
+                        },
+                    },
+                },
+                development: {
+                    name: 'development',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {
+                        bar: {
+                            name: 'bar',
+                            version: 1337,
+                            queuedVersion: 0,
+                            undeployVersion: false,
+                            locks: {
+                                locktest: {
+                                    message: 'locktest',
+                                    lockId: 'ui-v2-1337',
+                                    createdAt: new Date(1995, 11, 17),
+                                },
+                                lockbar: { message: 'lockbar', lockId: 'ui-v2-321', createdAt: new Date(1995, 11, 15) },
+                            },
+                        },
+                    },
+                },
+            },
+            filter: 'bar',
+            expectedLockIDs: ['ui-v2-321', 'ui-v2-1337'],
+        },
+    ];
+
+    describe.each(sampleAppFilteredData)(`Test Filtered Lock IDs`, (testcase) => {
+        it(testcase.name, () => {
+            // given
+            UpdateOverview.set({ environments: testcase.envs });
+            // when
+            const obtained = renderHook(() => useFilteredApplicationLockIDs(testcase.filter)).result.current;
```

Comment:
the goal of this PR is not clear to me.
Are all of these changes necessary to avoid the react error message?

Reply:
The errors are due to the way we currently get the locks (as a `DisplayLock`) from the store. For the mapper to be able to produce shallow comparable results, the array that is mapped must be a primitive type or a very simple object (with just primitive types in the "1st level").
example: `{key: string; id: number}` works, `{key: string; date: Date}` doesn't work because `Date` is also an object

So my plan was to create functions in the store that would return just strings (lock IDs). Those strings would go into the map, the map would be able to produce shallow comparable results because of the simple strings. Then the strings would be used to get the full object from the store to be able to display it.